python/python/bystro/prs/model.py

```diff
@@ -0,0 +1,145 @@
+import logging
+import os
+from pathlib import Path
+import requests
+
+from msgspec import Struct
+
+from bystro.utils.timer import Timer
+
+logging.basicConfig(
+    filename="prs_model.log",
+    level=logging.DEBUG,
+    format="%(asctime)s.%(msecs)03d %(levelname)s %(module)s - %(funcName)s: %(message)s",
+    datefmt="%Y-%m-%d %H:%M:%S",
+)
+logger = logging.getLogger()
+
+PRS_BUCKET = os.getenv("PRS_BUCKET", "bystro-prs-public")
+PRS_MODEL_DIR = Path(os.getenv("PRS_MODEL_DIR", str(Path(__file__).parent / "data")))
+PRS_MODEL_MAP_SUBDIR = "map"
+PRS_MODEL_SUMSTATS_SUBDIR = "sumstats"
+
+
+class PrsModel(Struct, frozen=True, forbid_unknown_fields=True, rename="camel"):
+    map_path: str
+    score_path: str
+
+
+models_cache: dict[str, PrsModel] = {}
+map_cache = {}
+sumstats_cache = {}
+
+map_template = "%s_genetic_map_%s.feather"
+sumstats_template = "%s_sumstats_%s_%s_compressed.feather"
+
+
+def download_file(bucket: str, key: str, filename: str):
+    """
+    Download a file from the given URL to the local path.
+
+    Args:
+        url (str): The URL to download the file from.
+        local_path (str): The local path to save the file.
+    """
+    url = f"https://{bucket}.s3.amazonaws.com/{key}"
+    logger.info("Downloading file from %s to %s", url, filename)
+    response = requests.get(url)
+    response.raise_for_status()  # Raise an error for bad status codes
+
+    with open(filename, "wb") as f:
+        f.write(response.content)
+
+
+def get_sumstats_file(disease: str, assembly: str, pmid: str) -> Path:
+    """
+    Get the path to the sumstats file for a given disease, assembly, and PubMed ID.
+
+    Args:
+        disease (str): The disease to use.
+        assembly (str): The assembly to use.
```

Comment:
Another thing to keep in mind is that GWAS sum stats might not have the same assembly as the target/annotated dataset, unless we're personally lifting them over so we have both options or requesting them to be lifted over by the user and we're ingesting that version